src/charm.py

```diff
@@ -176,9 +149,24 @@ def _check_model_name(self):
 
     def _check_leader(self):
         if not self.unit.is_leader():
-            # We can't do anything useful when not the leader, so do nothing.
             raise CheckFailed("Waiting for leadership", WaitingStatus)
 
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+        current_layer = self.container.get_plan()
+        new_layer = self._kubeflow_dashboard_operator_layer
+        self.logger.debug(f"NEW LAYER: {new_layer}")
+        if current_layer.services != new_layer.services:
+            self.unit.status = MaintenanceStatus("Applying new pebble layer")
+            self.container.add_layer(self._container_name, new_layer, combine=True)
+            try:
+                self.logger.info(
+                    "Pebble plan updated with new configuration, replaning"
+                )
+                self.container.replan()
+            except ChangeError:
+                raise CheckFailed("Failed to replan", BlockedStatus)
```

Comment:
afaict, this helper is reusable boilerplate.  I've just reviewed the same code in the [kserve sidecar charm](https://github.com/canonical/kserve-operators/pull/2/files#diff-ad55a839047964c277b40ec9dea61ff441e4a7bb59b06de9d159673f6d84ef75R78).  Let's refactor this to be an encapsulated, reusable helper and put it in chisme

(note: I left the same comment in the other charm, too)

Reply:
Can we open a task for this (this one is already way too broad IMO) :)

Reply:
Fully agree, opened a jira ticket for it next sprint, also mentioned in the ticket that we should update kserve and dashboard once it is done. Lets not block this PR on it though.